### PR TITLE
DEVXP-2811 - Add precommit

### PR DESCRIPTION
**Pull request by vschoener**

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        types: [python]
+        require_serial: true
+
+  - repo: local
+    hooks:
+      - id: spectral-lint
+        name: spectral-lint
+        entry: make spectral-lint
+        files: 'docs/openapi/.*\.yaml'
+        language: system
```
Comment (vitorquintanilha) [inline .pre-commit-config.yaml]: I don't think we need this

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    hooks:
+      - id: check-toml
+      - id: trailing-whitespace
+      - id: end-of-file-fixer
+
+  - repo: https://github.com/shellcheck-py/shellcheck-py
+    rev: 'v0.10.0.1'
+    hooks:
+      - id: shellcheck
+        args: ['-x']
```
Comment (vitorquintanilha) [inline .pre-commit-config.yaml]: I don't think we need this ^^

File: libyear/export_format/__init__.py [hunk 3]
```diff
@@ -0,0 +1 @@
+name = "libyear"
```
Comment (vitorquintanilha) [inline libyear/export_format/__init__.py]: What is this for? ^^

File: pyproject.toml [hunk 4]
```diff
@@ -34,3 +36,109 @@ build-backend = "poetry.core.masonry.api"
...
 license = {text = "MIT License"}
+
+[tool.poetry.group.typing.dependencies]
+mypy = "1.9.0"
+
+[tool.bandit]
+exclude_dirs = ["tests"]
+
+[tool.coverage.run]
+source = ["upload_service_cl"]
```
Comment (vitorquintanilha) [inline pyproject.toml]: Upsie

File: pyproject.toml [hunk 5]
```diff
@@ -34,3 +36,109 @@ build-backend = "poetry.core.masonry.api"
...
+    # Coverage is not expected on pytest.fail() calls in tests
+    ".*pytest.fail.*",
+]
+fail_under = 80
+skip_covered = true
+show_missing = true
+
+[tool.mypy]
+python_version = "3.12"
+files = ["upload_service_cl"]
```
Comment (vitorquintanilha) [inline pyproject.toml]: Upsie

File: pyproject.toml [hunk 6]
```diff
@@ -34,3 +36,109 @@ build-backend = "poetry.core.masonry.api"
...
+ignore_missing_imports = true
+
+[tool.pytest.ini_options]
+# Make sure that warnings raise errors, to handle things like deprecations as soon as possible.
+filterwarnings = [
+    "error",  # Make sure that warnings raise errors, to handle things like deprecations as soon as possible.
+    # overrides for the errors: schemathesis and hypothesis
+    "ignore::DeprecationWarning:hypothesis_jsonschema",
+    "ignore::DeprecationWarning:schemathesis",
+    "ignore:.*custom tp_new.*in Python 3.14.*:DeprecationWarning", # This is an issue not resolved in the gcs lib yet
```
Comment (vitorquintanilha) [inline pyproject.toml]: I don't think we need those exceptions here

File: pyproject.toml [hunk 7]
```diff
@@ -34,3 +36,109 @@ build-backend = "poetry.core.masonry.api"
...
+force-single-line = true
+
+[tool.ruff.per-file-ignores]
+# Some modules are not part of the service package, so it's ok not to have an `__init__.py`,
+# which is why we ignore `INP001` (because it's false positives for those specific cases for us).
+"conf/*" = ["INP001"]
+"scripts/*" = ["INP001"]
+
+[tool.ruff.flake8-bugbear]
+extend-immutable-calls = ["fastapi.Depends", "fastapi.params.Depends"]
```
Comment (vitorquintanilha) [inline pyproject.toml]: I don't think we need this here ^^

File: libyear/export_format/__init__.py (hunk 3 above)
Comment (vschoener) [reply libyear/export_format/__init__.py]: copy/paste from another __init__ file (so lazy with them I just copy past one to another without paying attention :D)

File: pyproject.toml (hunk 4 above)
Comment (vschoener) [reply pyproject.toml]: Oops 🙊

File: pyproject.toml (hunk 6 above)
Comment (vschoener) [reply pyproject.toml]: You right

**Review by vitorquintanilha — APPROVED** [review]

**Pull request merged**